src/vipersci/vis/db/raw_products.py

```diff
@@ -0,0 +1,544 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Reuse is permitted under the terms of the license.
+# The AUTHORS file and the LICENSE file are at the
+# top level of this library.
+
+from datetime import datetime, timedelta, timezone
+from warnings import warn
+
+from sqlalchemy import orm
+from sqlalchemy.orm import synonym, validates
+from sqlalchemy import (
+    DateTime,
+    Integer,
+    String,
+    Column,
+    Boolean,
+    Float,
+    Identity,
+)
+from sqlalchemy.ext.hybrid import hybrid_property
+
+from vipersci.pds.pid import VISID, vis_instruments, vis_compression
+from vipersci.pds.datetime import isozformat
+from vipersci.vis.header import pga_gain as header_pga_gain
+
+
+Base = orm.declarative_base()
+
+
+class RawProduct(Base):
+    """An object to represent rows in the Raw_Products table for VIS.
+    """
+
+    # This class is derived from SQLAlchemy's orm.declarative_base()
+    # which means that it has a variety of class properties that are
+    # then swept up into properties on the instantiated object via
+    # super().__init__().
+
+    # The table represents many of these objects, so the __tablename__ is
+    # plural while the class name is singular.
+    __tablename__ = "raw_products"
+
+    # The Column() names below should use "snake_case" for the names that are
+    # committed to the database as column names.  Furthermore, those names
+    # should be similar, if not identical, to the PDS4 Class and Attribute
+    # names that they represent.  Other names (like Yamcs parameter camelCase
+    # names) are implemented as synonyms. Aside from the leading "id" column,
+    # the remainder are in alphabetical order, since there are so many.
+
+    id = Column(Integer, Identity(start=1), primary_key=True)
+    adc_gain = Column(
+        Integer, nullable=False, doc="ADC_GAIN from the MCSE Image Header."
+    )
+    adcGain = synonym("adc_gain")
+    auto_exposure = Column(
+        Boolean,
+        nullable=False,
+        doc="AUTO_EXPOSURE from the MCSE Image Header.",
+    )
+    autoExposure = synonym("auto_exposure")
+    bad_pixel_table_id = Column(
+        Integer,
+        nullable=False,
+        # There is a Defective Pixel Map (really a list of 128 coordinates) for
+        # each MCAM.  The "state" of this is managed by the ground and not
+        # reflected in any Image Header information attached to an individual
+        # image.  It is not clear how to obtain this information from Yamcs,
+        # or even what might be recorded, so this column's value is TBD.
+    )
+    cameraId = synonym(
```

Comment:
Yeah, I waffled about that, but came down on the alphabetize everything here side.

Reply:
My reason for suggesting that the synonyms stay with the original columns is that some columns don't have synonyms and so it may not be immediately obvious when looking at this API that a particular column has a synonym.

Reply:
This is a difference of opinion. I think the use case for putting eyeballs on this code is to find an entry, and that's easiest if things are alphabetical.  If you need to know if a column has a synonym, you can programmatically query it, or search for its string in the file, and I don't think that you're even going to do that terribly often.

I'm not going to re-order these.  After this PR lands, if it really bothers you, you can re-order them and submit a PR.

Reply:
No. It's an unimportant distinction overall.